includes/embeds/class-amp-facebook-embed.php

```diff
@@ -131,12 +79,20 @@ public function sanitize_raw_embeds( Document $dom ) {
 		 */
 		$fb_root = $dom->getElementById( 'fb-root' );
 		if ( $fb_root ) {
-			$scripts = [];
-			foreach ( $dom->xpath->query( '//script[ starts-with( @src, "https://connect.facebook.net" ) and contains( @src, "sdk.js" ) ]' ) as $script ) {
-				$scripts[] = $script;
-			}
-			foreach ( $scripts as $script ) {
-				$script->parentNode->removeChild( $script );
+			/**
+			 * Script.
+			 *
+			 * @var DOMElement $script
+			 */
+			$script_query = $dom->xpath->query( '//script[ starts-with( @src, "https://connect.facebook.net" ) and contains( @src, "sdk.js" ) ]' );
+			foreach ( $script_query as $script ) {
+				$parent_node = $script->parentNode;
+				$parent_node->removeChild( $script );
+
+				// Remove parent node if it is an empty <p> tag.
+				if ( 'p' === $parent_node->nodeName && null === $parent_node->firstChild ) {
+					$parent_node->parentNode->removeChild( $parent_node );
+				}
 			}
 			$fb_root->parentNode->removeChild( $fb_root );
```

Comment:
It turns out that when you have multiple Facebook embeds on a page, each instance will include a `div#fb-root`. So I suppose this should be changed to:

```suggestion
			while ( $fb_root ) {
				$fb_root->parentNode->removeChild( $fb_root );
				$fb_root = $dom->getElementById( 'fb-root' );
			}
```

Reply:
In theory this should work, but when it retrieves the `#fb_root` `div` again it's returning the same `div` as before. I'll adapt this to use an XPath query instead.

Reply:
Why would it return the same `div` as before if it was removed from the document?